CMakeLists.txt

```diff
@@ -932,12 +933,11 @@ set(QGIS_INSTALL_SYS_LIBS TRUE CACHE BOOL "If set to TRUE install all required s
 
 #############################################################
 # Python build dependency
-
-find_package(PythonLibrary REQUIRED)
-set(MIN_PYTHON_VERSION "3.7")
-if(${PYTHON_SHORT_VERSION} VERSION_LESS ${MIN_PYTHON_VERSION})
-    message(FATAL_ERROR "Python version ${PYTHON_SHORT_VERSION} is too old. Minimum Python version is ${MIN_PYTHON_VERSION}.")
-endif()
+set(Python_FIND_FRAMEWORK "LAST")
```

Comment:
I will have to run, but I think you may need to set Python_ROOT_DIR on mac to "QGIS_MAC_DEPS_DIR" or similar. The key is that on mac the python from qgis-deps is picked BEFORE any system or homebrew or custom installation. otherwise it will not compile or have any random crashes.

Reply:
it's done with this policy: https://cmake.org/cmake/help/git-stage/policy/CMP0094.html